clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy

```diff
@@ -177,14 +186,105 @@ class KubernetesClientApiAdapter {
     }
   }
 
+  V1beta1StatefulSet replaceStatfulSet(String name, String namespace, V1beta1StatefulSet statefulSet) {
+    exceptionWrapper("statefulSets.replace", "Replace Stateful Set ${name}", namespace) {
+      def deployedControllerSet = getStatefulSet(name, namespace)
+      return apiInstance.replaceNamespacedStatefulSet(name, namespace, KubernetesClientApiConverter.toReplaceStatefulSet(statefulSet, deployedControllerSet), API_CALL_RESULT_FORMAT)
+    }
+  }
+
+  V1beta1StatefulSet resizeStatefulSet(String name, String namespace, int targetSize) {
+    exceptionWrapper("statefulSets.resize", "Resize Stateful Set $name", namespace) {
+      KubernetesControllerPatch operation = new KubernetesControllerPatch()
+      operation.op = "replace"
+      operation.path = "/spec/replicas"
+      operation.value = targetSize
+      List<KubernetesControllerPatch> patch = new ArrayList<KubernetesControllerPatch>()
+      patch.add(operation)
+
+      V1beta1StatefulSet statefulSet = apiInstance.patchNamespacedStatefulSet(name, namespace, patch, null)
+      def labels = statefulSet.metadata.labels
+      V1PodList pods = getPods(namespace, labels)
+      int podSize = pods.items.size()
+      int terminationGracePeriodSeconds = statefulSet.spec.template.spec.terminationGracePeriodSeconds ?: TERMINATION_GRACE_PERIOD_SECONDS
+
+      if (podSize != targetSize) {
+        waitForNumberOfPodUpdated(namespace, labels, podSize, targetSize, terminationGracePeriodSeconds)
```

Comment:
This shouldn't happen here - orca should be handling the "wait for up instances"

Reply:
I can skip this logic for resize, but it needs for "update strategies" because I need to ensure all pods down before I can update the new statefulset and daemonset definition.